tests/system/python/api/test_service.py

```diff
@@ -0,0 +1,300 @@
+        assert 'FogLAMP Core' == core_svc['name']
+        assert 'http' == core_svc['protocol']
+
+    @pytest.mark.parametrize("plugin, svc_name, display_svc_name, config, enabled, svc_count", [
+        ("randomwalk", SVC_NAME_1, SVC_NAME_1, None, True, 3),
+        ("http_south", SVC_NAME_2, SVC_NAME_1, None, False, 3),
+        ("Benchmark", SVC_NAME_3, SVC_NAME_3, None, True, 4),
+        ("Random", SVC_NAME_4, SVC_NAME_3, None, False, 4)
+    ])
+    def test_add_service(self, foglamp_url, wait_time, plugin, svc_name, display_svc_name, config, enabled, svc_count):
+        jdoc = plugin_and_service.add_south(plugin, foglamp_url, svc_name, config, enabled)
+        assert svc_name == jdoc['name']
+        assert UUID(jdoc['id'], version=4)
+
+        time.sleep(wait_time)
+        jdoc = get_service(foglamp_url, '/foglamp/service')
+        assert len(jdoc), "No data found"
+        assert svc_count == len(jdoc['services'])
+        southbound_svc = jdoc['services'][svc_count - 1]
+        assert isinstance(southbound_svc['management_port'], int)
+        assert southbound_svc['service_port'] is None
+        assert display_svc_name == southbound_svc['name']
+        assert 'running' == southbound_svc['status']
+        assert 'Southbound' == southbound_svc['type']
+        assert 'localhost' == southbound_svc['address']
+        assert 'http' == southbound_svc['protocol']
+
+    def test_add_service_with_config(self, foglamp_url, wait_time):
+        # add service with config param
+        data = {"name": SVC_NAME_5,
+                "type": "South",
+                "plugin": 'randomwalk',
+                "config": {"maxValue": {"value": "20"}, "assetName": {"value": "Random"}},
+                "enabled": True
+                }
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", '/foglamp/service', json.dumps(data))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert SVC_NAME_5 == jdoc['name']
+        assert UUID(jdoc['id'], version=4)
+
+        # verify config is correctly saved
+        conn.request("GET", '/foglamp/category/{}'.format(SVC_NAME_5))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert data['config']['assetName']['value'] == jdoc['assetName']['value']
+        assert data['config']['maxValue']['value'] == jdoc['maxValue']['value']
+
+        time.sleep(wait_time)
+        jdoc = get_service(foglamp_url, '/foglamp/service')
+        assert len(jdoc), "No data found"
+        assert 5 == len(jdoc['services'])
+        assert SVC_NAME_5 == jdoc['services'][4]['name']
+
+    @pytest.mark.parametrize("svc_name, status, svc_count", [
+        ("FogLAMP Storage", 404, 2),
+        ("FogLAMP Core", 404, 2),
+        (SVC_NAME_1, 200, 4),
```

Comment:
if it would have been me, I would broken this down into:

     def test_delete_core_and_storage_services
     def test_delete_service